client/sites/tools/logs/hooks/use-view.ts

```diff
@@ -0,0 +1,92 @@
+import { useState } from 'react';
+import { FilterType, LogType } from 'calypso/data/hosting/use-site-logs-query';
+import type { View } from '@wordpress/dataviews';
+
+const getSortField = ( logType: LogType ) => ( logType === LogType.PHP ? 'timestamp' : 'date' );
+const getVisibleFields = ( logType: LogType ) => {
+	if ( logType === LogType.PHP ) {
+		return [ 'severity', 'message' ];
+	}
+	return [ 'request_type', 'status', 'request_url' ];
+};
+const getFilterValue = ( view: View, fieldName: string ) =>
+	view.filters?.filter( ( filter ) => filter.field === fieldName )?.[ 0 ]?.value;
+
+function toFilterParams( { view, logType }: { view: View; logType: LogType } ): FilterType {
+	const filters: FilterType = {};
+
+	if ( logType === LogType.PHP ) {
+		const severity = getFilterValue( view, 'severity' );
+		if ( severity ) {
+			filters.severity = severity;
+		}
+	}
+
+	if ( logType === LogType.WEB ) {
+		const cached = getFilterValue( view, 'cached' );
+		const requestType = getFilterValue( view, 'request_type' );
```

Comment:
Nit: I guess you could iterate the array once and assign all values.

Reply:
Applied the suggestion at https://github.com/Automattic/wp-calypso/pull/99056/commits/c9409cf90595983bb3aea72118477b0bc6edd7ef